### PR TITLE
WEBDEV-6276 Added custom slot for result-cta-tiles

### PR DESCRIPTION
**Pull request by nsharma123**
https://webarchive.jira.com/browse/WEBDEV-6276

Description: let's add custom slot in collection-browser for result-cta-tile on ?tab=uploads and ?tab=web-archive tab so what we can a extra tile to upload new content.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
     "@internetarchive/histogram-date-range": "^1.1.0",
     "@internetarchive/ia-activity-indicator": "^0.0.4",
     "@internetarchive/ia-dropdown": "^1.1.0",
-    "@internetarchive/infinite-scroller": "^1.0.0",
+    "@internetarchive/infinite-scroller": "1.0.0-alpha7",
```
Comment (latonv) [inline package.json:32]: (Just a reminder to upgrade this off alpha before we merge in)

File: src/app-root.ts [hunk 2]
```diff
@@ -229,91 +229,93 @@ export class AppRoot extends LitElement {
...
-                  min="10"
-                  max="100"
-                  value="29"
-                  step="0.1"
-                  id="cell-height-slider"
-                  @input=${this.heightChanged}
-                />
-                <span>${this.cellHeight}rem</span>
-              </div>
+          <fieldset class="cell-controls">
```
Comment (latonv) [inline src/app-root.ts:232]: 👍

File: src/app-root.ts [hunk 3]
```diff
@@ -558,6 +575,26 @@ export class AppRoot extends LitElement {
...
+  private resultLastTileViewCheckboxChanged(e: Event) {
+    const target = e.target as HTMLInputElement;
+
+    const div = document.createElement('div');
+    const title = document.createElement('h3');
+    title.textContent = 'Upload';
+
+    div.setAttribute('slot', 'result-last-tile');
+    div.setAttribute('class', 'result-last-tile');
+    div.appendChild(title);
```
Comment (latonv) [inline src/app-root.ts:587]: Q: We still construct the new element even when the box is unchecked -- could we avoid this?

File: src/collection-browser.ts [hunk 4]
```diff
@@ -586,7 +586,10 @@ export class CollectionBrowser
       .placeholderCellTemplate=${this.placeholderCellTemplate}
       @scrollThresholdReached=${this.scrollThresholdReached}
       @visibleCellsChanged=${this.visibleCellsChanged}
-    ></infinite-scroller>`;
+      >${this.displayMode === 'grid'
+        ? html`<slot name="result-last-tile" slot="result-last-tile"></slot>`
+        : nothing}
```
Comment (latonv) [inline src/collection-browser.ts:591]: 👍

**Review by latonv — APPROVED** [review]
LGTM -- just had one small Q, no blockers.

File: src/app-root.ts (hunk 3 above)
Comment (nsharma123) [reply src/app-root.ts:587]: I don't think is construct without clicking on the checkbox. I verified with some console statement. Thannks!

**Pull request merged**